### PR TITLE
implemented cinema_visit function

### PR DESCRIPTION
**Pull request by BGBroth**

### CODE REVIEW

File: app/cinema/hall.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from app.people.cinema_staff import Cleaner
+
+
+class CinemaHall:
+    def __init__(self, number: int) -> None:
+        self.number = number
+
+    def movie_session(self,
+                      movie_name: str,
+                      customers: list,
```
Comment (Bohdan-Salamakha) [inline app/cinema/hall.py:10]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]

File: app/main.py [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+from app.cinema.bar import CinemaBar
+from app.cinema.hall import CinemaHall
+from app.people.customer import Customer
+from app.people.cinema_staff import Cleaner
 
 
-def cinema_visit(customers: list, hall_number: int, cleaner: str, movie: str):
-    # write you code here
-    pass
+def cinema_visit(customers: list,
```
Comment (Bohdan-Salamakha) [inline app/main.py:7]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

File: app/cinema/hall.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from app.people.cinema_staff import Cleaner
+from app.people.customer import Customer
+
+
+class CinemaHall:
+    def __init__(self, number: int) -> None:
+        self.number = number
+
+    def movie_session(self,
```
Comment (Polyakiv-Andrey) [inline app/cinema/hall.py:9]: After the opening bracket, write the argument on a new line and write the closing bracket in the same way on a new line. Do this always if you have 1 argument per line.

File: app/main.py [hunk 4]
```diff
@@ -1,6 +1,21 @@
...
+from app.cinema.bar import CinemaBar
+from app.cinema.hall import CinemaHall
+from app.people.customer import Customer
+from app.people.cinema_staff import Cleaner
 
 
-def cinema_visit(customers: list, hall_number: int, cleaner: str, movie: str):
-    # write you code here
-    pass
+def cinema_visit(customers: list[Customer],
```
Comment (Polyakiv-Andrey) [inline app/main.py:7]: After the opening bracket, write the argument on a new line and write the closing bracket in the same way on a new line. Do this always if you have 1 argument per line.

File: app/main.py [hunk 5]
```diff
@@ -1,6 +1,21 @@
...
+                 hall_number: int,
+                 cleaner: str,
+                 movie: str) -> None:
+    customers_inst = [Customer(name=customer["name"],
```
Comment (Polyakiv-Andrey) [inline app/main.py:11]: after opening the list generator, write the operation on a new line, the loop on a new line, and the closing parenthesis on a new line

**Review by Polyakiv-Andrey — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 6]
```diff
@@ -1,6 +1,24 @@
...
+from app.cinema.hall import CinemaHall
+from app.people.customer import Customer
+from app.people.cinema_staff import Cleaner
 
 
-def cinema_visit(customers: list, hall_number: int, cleaner: str, movie: str):
-    # write you code here
-    pass
+def cinema_visit(
+        customers: list[Customer],
```
Comment (Dimosphen1) [inline app/main.py:8]: Wrong type hinting, here we expect to get a list that consists of dicts, not `Customer` instances.

File: app/main.py [hunk 7]
```diff
@@ -1,6 +1,24 @@
...
+        hall_number: int,
+        cleaner: str,
+        movie: str
+) -> None:
+    customers_inst = [
+        Customer(name=customer["name"], food=customer["food"])
```
Comment (Dimosphen1) [inline app/main.py:14]: Just as an option, you could initialize the `Customer` class with the use of unpacking, as `Customer(**customer)`.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Minor change requested in type hinting.

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**